common/remote/sockfile.cpp

```diff
@@ -1892,8 +1903,30 @@ class CRemoteFilteredFileIOBase : public CRemoteBase, implements IFileIO
     {
         return request.append("\n }\n");
     }
+    void addVirtualFields(StringBuffer &request)
+    {
+        request.append(", \n \"virtualFields\" : {\n");
+        bool first=true;
+        for (auto &e : virtualFields)
+        {
+            if (!first)
+                request.append(",\n");
+            request.appendf("  \"%s\" : \"%s\"", e.first.c_str(), e.second.c_str());
+            first = false;
+        }
+        request.append(" }");
+    }
+    void handleFirstRequest()
+    {
+        firstRequest = false;
+        addVirtualFields(request);
```

Comment:
Slightly strange that request is a member, but is being passed to a member function.  Should the parameter for this and closeRequest() be removed?